# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientHandleParent.cpp
# Commit: dd31644a04d5
# Full Hash: dd31644a04d564ac7c0023fade672ac9222995de
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-10-01 21:34:52
# Description:
#   Bug 1578690 - Only create a new promise in EnsureSource if we didn't already have one. r=perry
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D47672
# ==============================================================================

diff -r f52bbe2a1903 -r dd31644a04d5 dom/clients/manager/ClientHandleParent.cpp
--- a/dom/clients/manager/ClientHandleParent.cpp	Tue Oct 01 08:27:56 2019 +0000
+++ b/dom/clients/manager/ClientHandleParent.cpp	Tue Oct 01 07:08:53 2019 +0000
@@ -77,7 +77,9 @@
     return SourcePromise::CreateAndResolve(mSource, __func__);
   }
 
-  mSourcePromise = new SourcePromise::Private(__func__);
+  if (!mSourcePromise) {
+    mSourcePromise = new SourcePromise::Private(__func__);
+  }
   return mSourcePromise;
 }
 
